runtime/src/main/java/org/corfudb/runtime/view/stream/AddressMapStreamView.java

```diff
@@ -159,16 +160,31 @@ protected boolean discoverAddressSpace(final UUID streamId,
                 queue.addAll(streamAddressSpace.copyAddressesToSet(maxGlobal));
 
                 long trimMark = streamAddressSpace.getTrimMark();
+
+                // In case we are dealing with a stream that does not have the checkpoint
+                // capability, check to see if we are trying to access an address that has been
+                // previously trimmed.
+                if (!checkpointCapableStream()
```

Comment:
What if the stream is transactional stream but trimMark >= stopAddress?

Reply:
This is exactly what this if clause is checking: if (transactionStream && trimMark >= stopAddress).